public/util/lex.ts

```diff
@@ -33,129 +75,288 @@ export function variablesToLexLanguage(variables: Variable[]): Lex {
+        filter.min = parseFloat(lq.min.key);
+        filter.max = parseFloat(lq.max.key);
+      } else {
+        filter.categories = [lq.value.key];
+      }
 
-    if (isNumericType(type)) {
-      filter.min = parseFloat(lq.min.key);
-      filter.max = parseFloat(lq.max.key);
-    } else if (type === DATETIME_FILTER) {
-      filter.min = dateToNum(lq.min);
-      filter.max = dateToNum(lq.max);
+      filters.push(filter);
     } else {
-      filter.categories = [lq.value.key];
-    }
+      const key = lq.field.key;
+      const type = lq.field.meta.type;
+      highlight = {
+        dataset,
+        context: "lex-bar",
+        key,
+        value: {},
+      } as Highlight;
 
-    return filter;
+      if (type === GEOBOUNDS_FILTER || type === GEOCOORDINATE_FILTER) {
+        highlight.key = highlight.key + "_group";
+        highlight.value.minX = parseFloat(lq.minX.key);
+        highlight.value.maxX = parseFloat(lq.maxX.key);
+        highlight.value.minY = parseFloat(lq.minY.key);
+        highlight.value.maxY = parseFloat(lq.maxY.key);
+      } else if (type === DATETIME_FILTER) {
+        highlight.value.min = dateToNum(lq.min);
+        highlight.value.max = dateToNum(lq.max);
+        highlight.type = DATETIME_FILTER;
+      } else if (isNumericType(type)) {
+        highlight.value.min = parseFloat(lq.min.key);
+        highlight.value.max = parseFloat(lq.max.key);
+        highlight.type = NUMERICAL_FILTER;
+      } else {
+        highlight.value = lq.value.key;
+      }
+    }
   });
-  return filters;
+  return {
+    filters: filters,
+    highlight: highlight,
+  };
+}
+
+function modeToRelation(mode: string): ValueStateValue {
+  switch (mode) {
+    case INCLUDE_FILTER:
+      return distilRelationOptions[0];
+    case EXCLUDE_FILTER:
+      return distilRelationOptions[1];
+    case HIGHLIGHT:
+      return distilRelationOptions[2];
+    default:
+      return distilRelationOptions[1];
+  }
 }
 
 function variablesToLexSuggestions(variables: Variable[]): ValueStateValue[] {
   if (!variables) return;
-  return variables.map((variable) => {
-    const name = variable.colDisplayName;
+
+  return variables.reduce((a, v) => {
+    const name = v.key;
     const options = {
-      type: colTypeToOptionType(variable.colType.toLowerCase()),
+      type: colTypeToOptionType(v.colType.toLowerCase()),
     };
-    return new ValueStateValue(name, options);
-  });
+    const config = {
+      displayKey: v.colDisplayName,
+    };
+    a.push(new ValueStateValue(name, options, config));
+
+    if (v.distilRole === "grouping") {
+      switch (v.colType) {
+        case TIMESERIES_TYPE:
+          const grouping = v.grouping as TimeseriesGrouping;
+          a.push(
+            new ValueStateValue(
+              grouping.xCol,
+              { type: DATETIME_FILTER },
+              { displayKey: grouping.xCol }
+            )
+          );
+          break;
+        case MULTIBAND_IMAGE_TYPE:
```

Comment:
They're explicitly passed over because we don't pull anything out of there grouping for now. Not necessarily to do, but I'll note that we don't want to fall to the unknown case because we know they're there, we're just not ungrouping anything from them for lex suggestions.